File: flutter_app/lib/config/tenant_config.dart
import 'package:flutter/material.dart';

enum LibraryFlavor {
  studyspace,
  libraryAbc,
  libraryXyz,
}

class TenantConfig {
  final LibraryFlavor flavor;
  final String tenantId;
  final String libraryName;
  final String libraryShortName;
  final String libraryTagline;
  final String libraryPhone;
  final String libraryEmail;
  final String libraryAddress;
  final double libraryLat;
  final double libraryLng;
  final double geofenceRadiusMeters;
  final Color primaryColor;
  final String footerCredit;
  final IconData libraryIcon;
  final String apiBaseUrl;

  const TenantConfig({
    required this.flavor,
    required this.tenantId,
    required this.libraryName,
    required this.libraryShortName,
    required this.libraryTagline,
    required this.libraryPhone,
    required this.libraryEmail,
    required this.libraryAddress,
    required this.libraryLat,
    required this.libraryLng,
    required this.geofenceRadiusMeters,
    required this.primaryColor,
    required this.footerCredit,
    required this.libraryIcon,
    required this.apiBaseUrl,
  });

  static TenantConfig _current = studyspaceConfig;

  static TenantConfig get current => _current;

  static void initialize(LibraryFlavor flavor) {
    switch (flavor) {
      case LibraryFlavor.studyspace:
        _current = studyspaceConfig;
        break;
      case LibraryFlavor.libraryAbc:
        _current = libraryAbcConfig;
        break;
      case LibraryFlavor.libraryXyz:
        _current = libraryXyzConfig;
        break;
    }
  }

  // Pre-configured Tenant Settings
  static const TenantConfig studyspaceConfig = TenantConfig(
    flavor: LibraryFlavor.studyspace,
    tenantId: 'studyspace_keshav',
    libraryName: 'Keshav Library & Study Center',
    libraryShortName: 'StudySpace',
    libraryTagline: 'SELF STUDY HALL',
    libraryPhone: '+91 98765 43210',
    libraryEmail: 'support@studyspace.com',
    libraryAddress: 'Near City Park, Main Road',
    libraryLat: 28.0087395,
    libraryLng: 73.2924508,
    geofenceRadiusMeters: 50.0,
    primaryColor: Color(0xFF1D4ED8),
    footerCredit: 'Ramxonwebwork',
    libraryIcon: Icons.menu_book_rounded,
    apiBaseUrl: 'https://library-management-hmwx.onrender.com',
  );

  static const TenantConfig libraryAbcConfig = TenantConfig(
    flavor: LibraryFlavor.libraryAbc,
    tenantId: 'library_abc',
    libraryName: 'ABC Competition Study Library',
    libraryShortName: 'ABC Library',
    libraryTagline: 'PREMIER READING HALL',
    libraryPhone: '+91 98000 11111',
    libraryEmail: 'contact@abclibrary.com',
    libraryAddress: '123 Academic Block, Zone 1',
    libraryLat: 19.0760,
    libraryLng: 72.8770,
    geofenceRadiusMeters: 50.0,
    primaryColor: Color(0xFF059669),
    footerCredit: 'Ramxonwebwork',
    libraryIcon: Icons.local_library_rounded,
    apiBaseUrl: 'https://library-abc-service-placeholder.onrender.com',
  );

  static const TenantConfig libraryXyzConfig = TenantConfig(
    flavor: LibraryFlavor.libraryXyz,
    tenantId: 'library_xyz',
    libraryName: 'XYZ Exam Preparation Hall',
    libraryShortName: 'XYZ Library',
    libraryTagline: '24/7 STUDY CENTRE',
    libraryPhone: '+91 97000 22222',
    libraryEmail: 'help@xyzlibrary.com',
    libraryAddress: '456 Knowledge Hub, Sector 4',
    libraryLat: 28.6139,
    libraryLng: 77.2090,
    geofenceRadiusMeters: 50.0,
    primaryColor: Color(0xFF7C3AED),
    footerCredit: 'Ramxonwebwork',
    libraryIcon: Icons.school_rounded,
    apiBaseUrl: 'https://library-xyz-service-placeholder.onrender.com',
  );
}

File: config/tenant_config.php
<?php
// config/tenant_config.php - Centralized Tenant Environment Configuration

if (!defined('LIBRARY_NAME')) {
    define('LIBRARY_NAME', getenv('LIBRARY_NAME') ?: 'Keshav Library & Study Center');
}

if (!defined('LIBRARY_SHORT_NAME')) {
    define('LIBRARY_SHORT_NAME', getenv('LIBRARY_SHORT_NAME') ?: 'StudySpace');
}

if (!defined('LIBRARY_TAGLINE')) {
    define('LIBRARY_TAGLINE', getenv('LIBRARY_TAGLINE') ?: 'SELF STUDY HALL');
}

if (!defined('LIBRARY_PHONE')) {
    define('LIBRARY_PHONE', getenv('LIBRARY_PHONE') ?: '+91 98765 43210');
}

if (!defined('LIBRARY_EMAIL')) {
    define('LIBRARY_EMAIL', getenv('LIBRARY_EMAIL') ?: 'support@studyspace.com');
}

if (!defined('LIBRARY_ADDRESS')) {
    define('LIBRARY_ADDRESS', getenv('LIBRARY_ADDRESS') ?: 'Near City Park, Main Road');
}

if (!defined('LIBRARY_LAT')) {
    define('LIBRARY_LAT', (float)(getenv('LIBRARY_LAT') ?: 28.0087395));
}

if (!defined('LIBRARY_LNG')) {
    define('LIBRARY_LNG', (float)(getenv('LIBRARY_LNG') ?: 73.2924508));
}

if (!defined('GEOFENCE_RADIUS_METERS')) {
    define('GEOFENCE_RADIUS_METERS', (float)(getenv('GEOFENCE_RADIUS_METERS') ?: 50.0));
}

if (!defined('PRIMARY_COLOR')) {
    define('PRIMARY_COLOR', getenv('PRIMARY_COLOR') ?: '#1D4ED8');
}

if (!defined('FOOTER_CREDIT')) {
    define('FOOTER_CREDIT', getenv('FOOTER_CREDIT') ?: 'Ramxonwebwork');
}

if (!defined('LIBRARY_ICON_CLASS')) {
    define('LIBRARY_ICON_CLASS', getenv('LIBRARY_ICON_CLASS') ?: 'fas fa-book-reader');
}
